### PR TITLE
572_bds_finding BDS Finding Vignette updated

### PR DESCRIPTION
**Pull request by millerg23**
Updated to match ADVS script updates and Best Practices.

### CODE REVIEW

File: vignettes/bds_finding.Rmd [hunk 1]
```diff
@@ -288,25 +341,30 @@ adeg <- derive_param_qtcf(
  
 ```
 
-```{r eval=TRUE, echo=FALSE}
-advs <- filter(advs, !(PARAMCD %in% c("MAP", "BSA")))
+All parameters now added so can add other parameter information from lookup
+table containing ``PARAMCD`` information.
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:345]: Here also i would rephrase: When all `PARAMCD` have been derived and added to the dataset, the other information from the look-up table (PARAM, PARAMCAT1,...) could be added.

File: vignettes/bds_finding.Rmd [hunk 2]
```diff
@@ -522,14 +607,19 @@ advs <- derive_extreme_flag(
...
 
 Another common example would be flagging the worst value for a subject,
 parameter, and visit. For this example, we will assume we have 3 `PARAMCD` 
-values (SYSBP, DIABP, and RESP).  We will also assume high is worst for SYSBP 
-and DIABP and low is worst for RESP.
+values (`SYSBP`, `DIABP`, and `RESP`).  We will also assume high is worst for `SYSBP` 
+and `DIABP` and low is worst for `RESP`.
 
 ```{r eval=TRUE}
 advs <- derive_extreme_flag(
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:625]: a new function has been added to handle this: `derive_worst_flag()`

File: vignettes/bds_finding.Rmd [hunk 3]
```diff
@@ -588,14 +686,71 @@ advs <- derive_obs_number(
...
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(advs, 
+                 display_vars = vars(USUBJID, PARAMCD, AVAL, AVALCA1N, AVALCAT1),
+                 filter = PARAMCD == "HEIGHT"
+)
 ```
 
-### Derive New Rows {#additional}
+## Add ADSL variables `ASEQ`](#adsl_vars)
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:735]: typo on the tag

File: vignettes/bds_finding.Rmd [hunk 4]
```diff
@@ -611,7 +766,7 @@ ad_advs.R | Derive a new PARAMCD based on existing PARAMCDs (MAP, based on Systo
 Example 1 (Creating a new record):
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:766]: Could you please set it at level 3: ###?
Also I would add more details on what record is added. like: In this example, a new record is created or each subject and parameter: AVAL is set to the last value on treatment, AVISIT is set to "End of Treatmenet " and AVISISTN...

File: vignettes/bds_finding.Rmd [hunk 5]
```diff
@@ -626,92 +781,38 @@ advs <- advs %>%
...
-  head(n = 10)
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(advs_ex1, 
+                 display_vars = vars(USUBJID, PARAMCD, ADT, AVISITN, AVISIT, ATPTN, AVAL, ANL01FL),
+                 filter = USUBJID == "01-701-1015" & ANL01FL == "Y"
+)
 ```
 
+
 Example 2 (Deriving a Summary record):
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:794]: same comment as above

File: vignettes/bds_finding.Rmd [hunk 6]
```diff
@@ -626,92 +781,38 @@ advs <- advs %>%
...
   by_vars = vars(STUDYID, USUBJID, PARAMCD, VISITNUM, ADT),
   analysis_var = AVAL,
   summary_fun = mean,
   set_values_to = vars(DTYPE = "AVERAGE")
 )
-
-head(select(advs, USUBJID, PARAMCD, VISITNUM, ADT, AVAL, DTYPE))
-```
-
-Example 3 (Derive a new PARAMCD):
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:651]: I think it would be good to show an example on how to use the derive_derived_param function. Even if not explicitely called: it may be good to show an example: Maybe with another MAP formula?
```
derive_derived_param(
   advs,
   by_vars = vars(USUBJID, VISIT),
   parameters = c("SYSBP", "DIABP"),
   analysis_value = (AVAL.SYSBP-AVAL.DIABP)/3 + AVAL.DIABP ,
   set_values_to = vars(
     PARAMCD = "MAP2",
     PARAM = "Mean Arterial Pressure 2 (mmHg)"
   )
 )

```

File: vignettes/bds_finding.Rmd [hunk 7]
```diff
@@ -626,92 +781,38 @@ advs <- advs %>%
...
-  left_join(avalcat_lookup, by = c("PARAMCD", "AVALCA1N"))
 
-advs %>% 
-  select(USUBJID, PARAMCD, AVAL, AVALCA1N, AVALCAT1) %>% 
-  filter(PARAMCD == "HEIGHT") %>% 
-  head()
-```
 
-### Example Scripts {#example}
+## Example Scripts {#example}
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:815]: should be level1 #

File: vignettes/bds_finding.Rmd (hunk 4 above)
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:766]: Just wondering if we need the sample code/description if we list the example below?

**Review by kabis-ops — COMMENTED** [review]
HI Gordon, thanks, just finished my review.

File: vignettes/bds_finding.Rmd [hunk 8]
```diff
@@ -26,13 +26,13 @@ result is captured in an SDTM Finding domain on a single date and/or time.
...
 
-## Programming Workflow
+# Programming Workflow
 
 * [Read in Data](#readdata)
 * [Derive/Impute Numeric Date/Time and Analysis Day (`ADT`, `ADTM`, `ADY`, `ADTF`, `ATMF`)](#datetime)
 * [Assign `PARAMCD`, `PARAM`, `PARAMN`, `PARCATy`](#paramcd)
 * [Derive Results (`AVAL`, `AVALC`, `AVALU`)](#aval)
-* [Derive Additional Parameters (e.g. `BMI` or `MAP` for ADVS)](#derive_param)
+* [Derive Additional Parameters (e.g. `BMI` `BSA` `MAP` for ADVS)](#derive_param)
```
[the remark this answers is not in the document]
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd [hunk 9]
```diff
@@ -41,15 +41,16 @@ otherwise specified.*
 * [Derive Analysis Flags (e.g. `ANL01FL`)](#analysisrec)
 * [Assign Treatment (`TRTA`, `TRTP`)](#treatment)
 * [Assign `ASEQ`](#aseq)
-* [Derive New Rows](#additional)
 * [Derive Categorization Variables (`AVALCATx`)](#cat)
+* [Add ADSL variables `ASEQ`](#adsl_vars)
+* [Derive New Rows](#additional)
 * [Example Scripts](#example)
```
[the remark this answers is not in the document]
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd [hunk 10]
```diff
@@ -41,15 +41,16 @@ otherwise specified.*
 * [Derive Analysis Flags (e.g. `ANL01FL`)](#analysisrec)
 * [Assign Treatment (`TRTA`, `TRTP`)](#treatment)
 * [Assign `ASEQ`](#aseq)
-* [Derive New Rows](#additional)
 * [Derive Categorization Variables (`AVALCATx`)](#cat)
+* [Add ADSL variables `ASEQ`](#adsl_vars)
```
[the remark this answers is not in the document]
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd [hunk 11]
```diff
@@ -196,88 +228,109 @@ param_lookup <- tribble(
   "TEMP", "TEMP", "Temperature (C)", 8, "Vital Sign", 2
 )
 attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+```
 
-head(param_lookup)
+We will create more parameters so only merge on `PARAMCD` for now from lookup 
+table. Then add all other derivations from lookup after all parameters created.
+
+```{r, eval=TRUE, echo=FALSE}
```
[the remark this answers is not in the document]
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd [hunk 12]
```diff
@@ -196,88 +228,109 @@ param_lookup <- tribble(
   "TEMP", "TEMP", "Temperature (C)", 8, "Vital Sign", 2
 )
 attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+```
 
-head(param_lookup)
+We will create more parameters so only merge on `PARAMCD` for now from lookup 
+table. Then add all other derivations from lookup after all parameters created.
+
```
[the remark this answers is not in the document]
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd (hunk 5 above)
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd (hunk 6 above)
Comment (millerg23) [reply vignettes/bds_finding.Rmd:651]: Done

File: vignettes/bds_finding.Rmd (hunk 7 above)
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd (hunk 4 above)
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Removed. Done.

File: vignettes/bds_finding.Rmd [hunk 13]
```diff
@@ -288,25 +341,30 @@ adeg <- derive_param_qtcf(
```
[the remark this answers is not in the document]
Comment (millerg23) [reply vignettes/bds_finding.Rmd:336]: Done

File: vignettes/bds_finding.Rmd (hunk 1 above)
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd (hunk 2 above)
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

File: vignettes/bds_finding.Rmd (hunk 3 above)
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Fixed.

File: vignettes/bds_finding.Rmd [hunk 14]
```diff
@@ -230,14 +229,10 @@ param_lookup <- tribble(
...
 
-We will create more parameters so only merge on `PARAMCD` for now from lookup 
-table. Then add all other derivations from lookup after all parameters created.
-
-```{r, eval=TRUE, echo=FALSE}
-dataset_vignette(param_lookup, 
-                 display_vars = vars(VSTESTCD, PARAMCD, PARAM, PARAMN, PARCAT1, 
-                                     PARCAT1N))
-```
+At this stage, only `PARAMCD` required to perform the derivations. Additional
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:232]: only PARAMCD **is** required

File: vignettes/bds_finding.Rmd [hunk 15]
```diff
@@ -810,9 +790,32 @@ dataset_vignette(advs_ex2,
 )
 ```
 
+### Example 3 (Deriving a new PARAMCD):
+Use function `derive_derived_param` to create a new `PARAMCD`. Below is an
+example of creating `Mean Arterial Pressure` (`PARAMCD`=`MAP`) with an
```
Comment (kabis-ops) [inline vignettes/bds_finding.Rmd:795]: MAP2

Comment (millerg23) [reply vignettes/bds_finding.Rmd:795]: Done

File: vignettes/bds_finding.Rmd (hunk 14 above)
Comment (millerg23) [reply vignettes/bds_finding.Rmd]: Done

**Review by kabis-ops — APPROVED** [review]
